compute/cker/include/cker/train/operation/Pad.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NNFW_CKER_TRAIN_OPERATION_PAD_H__
+#define __NNFW_CKER_TRAIN_OPERATION_PAD_H__
+
+#include "cker/operation/Pad.h"
+
+namespace nnfw
+{
+namespace cker
+{
+namespace train
+{
+
+/*
+ * input_data will be transformed by PAD operation with padding options(such as constant C) to
+ * output_data
+ *
+ * input_data  ->  output_data
+ *   [0,1]     ->    [C,C,C,C]
+ *   [2,3]     ->    [C,0,1,C]
+ *             ->    [C,2,3,C]
+ *             ->    [C,C,C,C]
+ */
+/*
+ * input_data(backward_output_data) will be transformed by backward of PAD operation (Depad) with
+ * padding options to output_data(backward_input_data)
+ *
+ * input_data(backward_output_data)  ->  output_data(backward_input_data)
+ *   [C,C,C,C]                       ->    [0,1]
+ *   [C,0,1,C]                       ->    [2,3]
+ *   [C,2,3,C]                       ->
+ *   [C,C,C,C]                       ->
+ */
+template <typename T>
+inline void Depad(const int32_t *padding_data, int32_t pad_rank, const Shape &input_shape,
+                  const T *input_data, const Shape &output_shape, T *output_data)
+{
+  using PaddingInfo = std::pair<int32_t, int32_t>;
+  using PaddingList = std::vector<PaddingInfo>;
+
+  assert(output_shape.DimensionsCount() == input_shape.DimensionsCount());
+
+  PaddingList padding_list(pad_rank);
+  for (int32_t n = 0; n < pad_rank; ++n)
+  {
+    const int32_t *from = padding_data + (n * 2);
+    padding_list[n] = {from[0], from[1]};
+  }
+  for (int32_t i = 0; i < pad_rank; ++i)
+  {
+    assert(output_shape.Dims(i) ==
+           input_shape.Dims(i) - padding_list[i].first - padding_list[i].second);
+  }
+
+  switch (pad_rank)
+  {
+    case 0:
+    case 1:
+    {
+      const int32_t out_width = output_shape.Dims(0);
+      const int32_t padding_left = padding_list[0].first;
+      std::memcpy(output_data, input_data + padding_left, out_width * sizeof(T));
```

Comment:
Does `input_data` and `output_data` memory layout is in order of `HWD`?

Reply:
That's a good point. Let me check it!

Reply:
After examining the Pad layer code, I have discovered that the Pad layer does not take into account any channel order of a rank 4 tensor but only implements NHWC tensor. It is necessary to add an assertion code that it must be NHWC.

Reply:
And @zetwhite discussed about padding rank, I realize that padding rank should be checkecd.

Reply:
Aha, My curiosity is somehow resolved by your offline explaination.

If `plan_rank` = 1 assure that 'input_shape rank is also 1', 
I think there is no problem with code :)

Reply:
I'll upload a new commit with two patches
1. check NHWC at rank-4
2. check pad rank

Reply:
> check NHWC at rank-4

I realize that this cannot be checked in cker level. I'll add code about that into backend code of PadLayer soon.